src/services/quandl-service.js

```diff
@@ -12,6 +12,31 @@
         QUANDL_URL = 'https://www.quandl.com/api/v3/',
         QUANDL_WIKI = 'datasets/WIKI/';
 
+    var useApiKey = true;
+
+    class Observable {
+        constructor() {
+            this.registry = {};
+        }
+
+        on(eventName, fn, scope) {
+            if (!this.registry[eventName]) {
+                this.registry[eventName] = [];
+            }
+
+            this.registry[eventName].push({
+                fn: fn,
+                scope: scope
+            });
+        }
+
+        notify(eventName, value) {
+            this.registry[eventName].forEach((entry) => {
+                entry.fn.call(entry.scope, value);
```

Comment:
Do we need to save the scopes and use `call`? Can we get around this with good use of lambdas and calling `entry.fn(value)` directly?